### PR TITLE
Skip setting RS485 if there is no user configuration

### PR DESCRIPTION
**Pull request by jjcarstens**
Previously we would try to set RS485 config if we were able to read it assuming that if you can read, you can set. However, its been discovered that Allwinner chips can return an RS485 configuration read but still fail to set.

This changes the logic to only attempt to set the RS485 config if the user has provided some config values and skip trying to set settings we just read in but don't actually need to be changed.

### CODE REVIEW

File: src/uart_comm_unix.c [hunk 1]
```diff
@@ -416,31 +416,36 @@ static void update_flags(uint32_t *flags, uint32_t flag_bit, int tristate_val)
...
+        struct serial_rs485 rs485;
 
-    if (ioctl(fd, TIOCGRS485, &rs485) < 0) {
-        // This may fail with ENOTTY if the port doesn't support RS485. If that's
-        // the case, then we only care about a failure when a user is trying
-        // to set things. Otherwise, we assume RS485 may not even be expected and
-        // can safely be ignored
-        if (!config->rs485_user_configured) return 0;
+        if (ioctl(fd, TIOCGRS485, &rs485) < 0) {
+            if (!config->rs485_user_configured) return 0;
```
Comment (fhunleth) [inline src/uart_comm_unix.c:428]:
```suggestion
```

This line can be deleted now.

Comment (jjcarstens) [reply src/uart_comm_unix.c]: Derp 🤦 - Removed

File: src/uart_comm_unix.c [hunk 2]
```diff
@@ -416,31 +416,34 @@ static void update_flags(uint32_t *flags, uint32_t flag_bit, int tristate_val)
...
 
-    if (ioctl(fd, TIOCSRS485, &rs485) < 0) {
-        record_errno();
-        return -1;
+        if (ioctl(fd, TIOCSRS485, &rs485) < 0) {
+            record_errno();
+            return -1;
+        }
+    } else {
+        return 0;
```
Comment (fhunleth) [inline src/uart_comm_unix.c:446]:
```suggestion
```

Comment (fhunleth) [reply src/uart_comm_unix.c:446]: That part can be deleted too.

**Review by fhunleth — APPROVED** [review]

**Comment by fhunleth** [comment]
Thanks!

**Pull request merged**